core/src/main/java/io/grpc/internal/TransportFrameUtil.java

```diff
@@ -61,7 +62,11 @@
    * @return the interleaved keys and values.
    */
   public static byte[][] toHttp2Headers(Metadata headers) {
-    byte[][] serializedHeaders = headers.serialize();
+    byte[][] serializedHeaders = InternalMetadata.serialize(headers);
+    // TODO(carl-mastrangelo): eventually remove this once all callers are updated.
+    if (serializedHeaders == null) {
+      return new byte[][]{};
```

Comment:
Maybe, but I plan on removing this line as soon as this PR is submitted.  Some test failed with this change, so I just put this in here to prevent pulling in more files.